### PR TITLE
GH_action/public_repo_tests_migration_b2

### PR DESCRIPTION
**Pull request by lim-deriv**
https://app.clickup.com/t/20696747/CICD-416

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        - 3.10.2
+        - 3.10.3
+        - 3.10.4
+        - 3.10.10
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pipenv'
```
Comment (nihal-deriv) [inline .github/workflows/build.yml:37]: does the setup python actions provide caching for the dependencies ? Is that why we pass cache argument ?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        - 3.9.16
+        - 3.10.0
+        - 3.10.1
+        - 3.10.2
+        - 3.10.3
+        - 3.10.4
+        - 3.10.10
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v4
```
Comment (nihal-deriv) [inline .github/workflows/build.yml:34]: are we missing the setting up the ssh agent step ?

File: .github/workflows/build.yml (hunk 1 above)
Comment (lim-deriv) [reply .github/workflows/build.yml:37]: Yes, I pass cache argument there is to cache the pipenv dependencies. https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md

File: .github/workflows/build.yml (hunk 2 above)
Comment (lim-deriv) [reply .github/workflows/build.yml:34]: This job did not have any command like git push or git clone so ssh agent is not needed.

**Review by nihal-deriv — APPROVED** [review]

**Pull request merged**